### ISSUE TITLE
Update and move frontend_views

### ISSUE THREAD

**Issue by gorkemarslan** [CiviWiki/OpenCiviWiki#1090]
The usage of the `frontent_views` folder is not compatible with the Django convention.

### Tasks

- [x] Reformat `frontend_views/views.py`
    - [x] If it is possible, use f-Strings
    - [x] Fix variable names. The name of the variables should be self-explanatory.
           **Examples of bad naming:**
https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L21
https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L100

- [x] `frontend_views/views.py` -> `accounts/views.py`
     - [x] [user_profile](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L61)
     - [x] Related URLs (`frontend_views/urls.py` -> `accounts/urls/urls.py`)

 - [x] `frontend_views/views.py` -> `threads/views.py`
     - [x] [base_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L17)
     - [x] [issue_thread](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L95)
     - [x] [civi2csv](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L189)
     - [x] Related URLs (`frontend_views/urls.py` -> `threads/urls.py`)

- [x] Discuss which app should include below:
    - [x] [create_group](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L161)
    - [x] [declaration](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L165)
    - [x] [landing_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L169)
    - [x] [how_it_works_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L173)
    - [x] [about_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L177)
    - [x] [support_us_view](https://github.com/CiviWiki/OpenCiviWiki/blob/d7f24fad7c0a2565da3bf2cd54e89a847d7479dd/project/frontend_views/views.py#L181)
    - [x] Except for `create_group`, it might be good to use [TemplateView](https://docs.djangoproject.com/en/3.2/ref/class-based-views/base/#templateview) for rest.
    - [x] Related URLs
    - [x] A good reference would be great to follow the Django convention about the views above. For example, `/about/` page does not belong to any model. We should decide where to put it. `thread/views.py` or a new file `core/views.py`?
  
- [x] Refactor for all the usages above. (including all `views.py` files and JS files)

- [x] Remove `frontend_views` folder

Since this is a long task and it requires some knowledge about the project. @AbhijithGanesh, I am assigning this issue.
Please start this issue after merging PR #1088 and #1089.

**Comment by AbhijithGanesh** [comment]
@gorkemarslan Why is the app frontend_views unconventional?

**Comment by AbhijithGanesh** [comment]
@brylie can you please confirm whether usage of frontend_views are unconventional?

**Comment by brylie** [comment]
@AbhijithGanesh the folder "frontend_views" is unconventional because all Django views are "frontend views" when Django is responsible for rendering templates. Basically, Django views should be defined within specific apps, such as `accounts` or `threads`, rather than a generic `frontend_views` app.

When the `frontend_views` app was created, the CiviWiki developers were working under the single-page application paradigm, where JavaScript would be responsible for rendering "front-end" content and Django would be relegated to serving JSON data via an HTTP API. The `frontend_views` app was seemingly created to contain some views that were rendered in a more idiomatic manner, using Django to render context data to an HTML template.

Basically, `frontend_views` app is a relic of early design decisions in the project. We are now taking the project in a different direction. Arguably, the recent changes follow more conventional Django development patterns and remove the overly complicated SPA architecture that wasn't a good fit for this project.

**Comment by gorkemarslan** [comment]
@AbhijithGanesh, the sub-task for the discussion is just for this purpose:

> Discuss which app should include below

> A good reference would be great to follow the Django convention about the views above

The one of the greatest activities to enhance your code quality is to look at the docs, to audit other open-source repos, etc. I just wonder here where and why we should store these files and learn your opinions about them. 

You can also look through another perspective. If you see a folder with name of `frontend_views`, what do you expect? This folder should include all views such as login, register, landing, etc. Right? But this is not the case. And in Django, views should be in the related apps. 

So we need to do something for `frontend_views`.

**Comment by AbhijithGanesh** [comment]
@gorkemarslan Can I start working on this again now? Or are we waiting for other PRs too?

**Comment by gorkemarslan** [comment]
> Please start this issue after merging PR #1088 and #1089.

Yes you can. :smile:

**Comment by gorkemarslan** [comment]
@AbhijithGanesh, how is this issue going?

Could you wait awhile for issue #1106 to be solved because that issue proposes some changes in `frontend_views` folder?

Thank you for patience. :blush:

**Comment by AbhijithGanesh** [comment]
I can wait for the same ☺️ don't worry Gorkem Arslan

**Comment by AbhijithGanesh** [comment]
@gorkemarslan Can I patch this today before we have any further changes?

**Comment by gorkemarslan** [comment]
@AbhijithGanesh, there are 2 opened PR working on `frontend_views`. I know you are eager to complete this task as soon as possible. Let's get @brylie 's opinion here.

@brylie, this issue is about removing `frontend_views` folder. But #1107 and #1110 are working on this folder. What should we do? 
If @AbhijithGanesh waits for these PRs, there won't any conflict. @AbhijithGanesh is ready to complete the tasks.
Or, if `frontend_views` folder is removed, I can make required changes PR #1110 and edit issue #1106.

**Comment by AbhijithGanesh** [comment]
Sure @gorkemarslan , next time make one mention and you can call me Abhijith if its hard typing my full name

**Comment by brylie** [comment]
I'd advise us to wait unless @AbhijithGanesh is comfortable syncing changes from `develop` to their feature branch.

**Comment by AbhijithGanesh** [comment]
I am ready to make a new branch to rework on this changes Brylie

**Comment by brylie** [comment]
Alright, just understand that you may need to merge changes from the `develop` branch into your new branch and resolve conflicts.

**Comment by AbhijithGanesh** [comment]
I am creating a totally new branch and every change will be integrated to the branch.

**Comment by AbhijithGanesh** [comment]
@gorkemarslan Which app should I add the miscellaneous functions too?(The last set of tasks)

**Comment by gorkemarslan** [comment]
@AbhijithGanesh, I guess you are talking about views functions, right? Then, you can put them into `threads` app.

**Comment by AbhijithGanesh** [comment]
I was talking about the view functions @gorkemarslan , I want to add them to core would that be fine?

**Comment by AbhijithGanesh** [comment]
@gorkemarslan why was this opened?

**Comment by gorkemarslan** [comment]
@AbhijithGanesh, as I explained in PR #1132:

> Apply changes in PR Routing issues #1116. In the current branch, some of these changes are gone somehow.
> Delete threads/view folder and move all content to threads/views.py file
> Remove frontend_views folder
> Delete unused and missing imports
> Add flake8 escape for django signals

**Comment by AbhijithGanesh** [comment]
Saw the PR @gorkemarslan
